opencart 1.5.6.4/upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -0,0 +1,333 @@
+<?php
+/**
+ * This is a plugin for Opencart to handle subscribers directly
+ * to your Smaily contacts, generate rss-feed of products and send
+ * abandoned cart emails with Smaily templates.
+ *
+ * @package smaily_for_opencart
+ * @author Smaily
+ * @license GPL-3.0+
+ * @copyright 2019 Smaily
+ *
+ * Plugin Name: Smaily for Opencart
+ * Description: Smaily email marketing and automation extension plugin for Opencart.
+ * Version: 1.1.3
+ * License: GPL3
+ * Author: Smaily
+ * Author URI: https://smaily.com/
+ *
+ * Smaily for Opencart is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * Smaily for Opencart is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Smaily for Opencart. If not, see <http://www.gnu.org/licenses/>.
+ */
+class ControllerModuleSmailyForOpencart extends Controller {
+    private $error = array();
+
+    public function index() {
+        // Add language file.
+        $this->load->language('module/smaily_for_opencart');
+        // Settings model.
+        $this->load->model('setting/setting');
+        // Smaily admin page model.
+        $this->load->model('smailyforopencart/admin');
+        // Add heading title.
+        $this->document->setTitle($this->language->get('heading_title'));
+        
+        // When save is pressed.
+        if (($this->request->server['REQUEST_METHOD'] == 'POST') && ($this->validate())) {
```

Comment:
This functionality is not used, as you do not have "Save" button under module settings. Along with block you can remove `validate` method (from this class), because it is not used by any other action.

Reply:
You did not remove `validate` method. Is it being used by something?